src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProject.cs

```diff
@@ -27,6 +28,21 @@ internal sealed partial class CPSProject : AbstractProject
             : base(projectTracker, reportExternalErrorCreatorOpt, projectDisplayName, projectFilePath,
                    hierarchy, language, projectGuid, serviceProvider, visualStudioWorkspaceOpt, hostDiagnosticUpdateSourceOpt, commandLineParserServiceOpt)
         {
+            if (binOutputPath != null)
+            {
+                // Ensure that binOutputPath is either null or a rooted path.
+                // CPS might provide such invalid paths during initialization or when project is in unrestored state.
+                if (binOutputPath == String.Empty)
+                {
+                    binOutputPath = null;
```

Comment:
Is null okay? I thought we blow up when the path is null - or is that only if was empty?